draft/0006-pat.md

```diff
@@ -0,0 +1,126 @@
+Host: tenant.logto.app
+Content-Type: application/json
+Authorization: Bearer <access-token-with-management-api-access>
+
+{
+  "name": "My PAT",
+}
+```
+
+```http
+HTTP/1.1 201 Created
+Content-Type: application/json
+
+{
+  "name": "My PAT",
+  "token": "pat_W51arOqe7nynW75nWhvYogyc",
+}
+```
+
+## Token Exchange with PAT
+
+Extend the token exchange flow to accept PATs as subject tokens and exchange them for short-lived access tokens.
+
+### Request
+
+The client makes a token exchange request to the tenant's token endpoint with a special grant type using the HTTP POST method. The following parameters are included in the HTTP request entity-body using the `application/x-www-form-urlencoded` format.
+
+1. `grant_type`: REQUIRED. The value of this parameter must be `urn:ietf:params:oauth:grant-type:token-exchange` indicates that a token exchange is being performed.
+2. `resource`: OPTIONAL. The resource indicator, the same as other token requests.
+3. `scope`: OPTIONAL. The requested scopes, the same as other token requests.
+4. `subject_token`: REQUIRED. The user's PAT.
+5. `subject_token_type`: REQUIRED. The type of the security token provided in the `subject_token` parameter. The value of this parameter must be `urn:logto:token-type:personal-access-token`.
+
+For security reasons, the client must authenticate with the token endpoint using the client ID and client secret in the `Authorization` header. Which means the client must be a traditional web application.
+
+And organization resources are not available for PATs.
+
+### Response
+
+If the token exchange request is successful, the tenant's token endpoint returns an access token that represents the identity of the user. The response includes the following parameters in the HTTP response entity-body using the `application/json` format.
+
+1. `access_token`: REQUIRED. The access token that represents the identity of the impersonated user.
+2. `issued_token_type`: REQUIRED. The type of the issued token. The value of this parameter must be `urn:ietf:params:oauth:token-type:access_token`.
+3. `token_type`: REQUIRED. The type of the token. The value of this parameter must be `Bearer`.
+4. `expires_in`: REQUIRED. The lifetime in seconds of the access token.
+5. `scope`: OPTIONAL. The scopes of the access token.
+
+### Example token exchange
+
+```http
+POST /oidc/token HTTP/1.1
+Host: tenant.logto.app
+Content-Type: application/x-www-form-urlencoded
+Authorization: Basic <base64(client-id:client-secret)>
+
+grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange
+&scope=profile
+&subject_token=pat_W51arOqe7nynW75nWhvYogyc
+&subject_token_type=urn%3Alogto%3Atoken-type%3Apersonal-access-token
+```
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "access_token": "eyJhbGci...zg",
+  "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+  "token_type": "Bearer",
+  "expires_in": 3600,
+  "scope": "profile"
+}
+```
+
```

Comment:
Do we need to describe common error scenarios, error response formats, and corresponding error messages?

Reply:
I don't think so, what you mentioned will be in openapi documents.